SimplenoteUITests/SimplenoteUITests.swift

```diff
@@ -0,0 +1,531 @@
+        continueAfterFailure = true
+        app.launch()
+        let _ = attemptLogOut()
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    func testLogInWithNoEmailNoPassword() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: "", password: "")
+
+        Assert.labelExists(labelText: text_LoginEmailInvalid)
+        Assert.labelExists(labelText: text_LoginPasswordShort)
+    }
+
+    func testLogInWithNoEmail() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: "", password: testDataExistingPassword)
+
+        Assert.labelExists(labelText: text_LoginEmailInvalid)
+        Assert.labelAbsent(labelText: text_LoginPasswordShort)
+    }
+
+    func testLogInWithInvalidEmail() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataInvalidEmail, password: testDataExistingPassword)
+
+        Assert.labelExists(labelText: text_LoginEmailInvalid)
+        Assert.labelAbsent(labelText: text_LoginPasswordShort)
+    }
+
+    func testLogInWithNoPassword() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: "")
+
+        Assert.labelAbsent(labelText: text_LoginEmailInvalid)
+        Assert.labelExists(labelText: text_LoginPasswordShort)
+    }
+
+    func testLogInWithTooShortPassword() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: testDataInvalidPassword)
+
+        Assert.labelAbsent(labelText: text_LoginEmailInvalid)
+        Assert.labelExists(labelText: text_LoginPasswordShort)
+    }
+
+    func testLogInWithExistingEmailIncorrectPassword() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: testDataNotExistingPassword)
+        Assert.alertExistsAndClose(headingText: text_AlertHeading_Sorry, content: text_AlertContent_LoginFailed, buttonText: uidButton_Accept)
+    }
+
+    func testLogInWithCorrectCredentials() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: testDataExistingPassword)
+        AllNotesAssert.screenShown()
+    }
+
+    func testLogOut() throws {
+        // Step 1
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: testDataExistingPassword)
+        AllNotesAssert.screenShown()
+
+        // Step 2
+        _ = logOut()
+        Assert.signUpLogInScreenShown()
+    }
+
+    // Test below is automatically generated one, disabled for now to save time
+    /*
+    func testLaunchPerformance() throws {
+        if #available(macOS 10.15, iOS 13.0, tvOS 13.0, *) {
+            // This measures how long it takes to launch your application.
+            measure(metrics: [XCTApplicationLaunchMetric()]) {
+                XCUIApplication().launch()
+            }
+        }
+    }*/
```

Comment:
Feel free to remove this. 

I think Xcode adds this to the template of a new UI tests bundle to show new developers how to write performance tests.

If we'll ever want to write performance tests in the UI test suite, we can easily find the [documentation online](https://developer.apple.com/documentation/xctest/xctestcase/1496290-measure).

Reply:
Went to 🗑️